### ISSUE TITLE
Possibly find a way to use thumbnails uploaded before PPP was active

### ISSUE THREAD

**Issue by cklosowski** [cklosowski/post-promoter-pro#113]
Since PPP registers it's own thumbnail sizes, images uploaded before PPP being active won't have the proper thumbnail sizes. We should see if we can find a way to to account for that.

**Comment by rumspeed** [comment]
Maybe this will help... http://www.deluxeblogtips.com/2015/01/resize-image-fly-wordpress.html

Possibly hook into the PPP scheduling action and if an image is selected from the form, check for your specific registered image name, run the function of not found.

Well, something like this. You will be the better judge of that.

**Comment by cklosowski** [comment]
Yeah that's a good idea. Better than what I was thinking, which was going to be more server intensive and run against every image. I'll look into doing it the way you suggested. Thanks @rumspeed

**Comment by rumspeed** [comment]
Cool. Glad my note helped. This might as well... I do remember the image in the schedule list showed the FULL IMAGE because it took a while to render at 5MB. At that point, it was already too late.Your fix for this will show your specific cropped image size (but fit in the display) and that will be a way to tell if it is working properly. That will be nice. Especially for people that install it after many posts have been created with images already in the media library prior to installing PPP. Let me know if you need help testing it.

Also, I think you can filter the media uploader to not show all available media sizes since you will be using your specific cropped image anyway. That will improve the user experience and set the expectation. They won't have to *think* about what size to use because you choose that for them.

Cheers!

**Comment by cklosowski** [comment]
They actually do'nt have to pick a size, i always use the PPP size if it's in the media uploader for the purpose of the best possible image cropped for the network.

**Comment by rumspeed** [comment]
Right on, I see what you mean and I like what it does. There is still an **option** to pick a size when using the file uploader.

In the "Scheduled Tweets" section when I click the "Insert File" button, it loads the "Insert Media" feature which has "Attachment Display Settings" that aren't necessary. I've seen this flow used before without the Alignment, Link To, and Size. PPP does't use those settings and I am pretty sure you can remove them and just use the "Insert into Post" button to select the image. From there, your system will handle the rest by using your custom media size.

![ppp-media-library](https://cloud.githubusercontent.com/assets/4145412/9059159/9a29641e-3a75-11e5-8d61-5cfeea510a65.jpg)

**Comment by cklosowski** [comment]
Oh nice, yeah i'll have to look into that for sure.

**Comment by cklosowski** [comment]
Thinking I can take this route, a stripped down version of what's used in the Regenerate Thumbnails plugin:
```
		// $id being the attachment ID
		$image = get_post( $id );
		$fullsizepath = get_attached_file( $image->ID );
		@set_time_limit( 900 ); // 5 minutes per image should be PLENTY
		$metadata = wp_generate_attachment_metadata( $image->ID, $fullsizepath );
		wp_update_attachment_metadata( $image->ID, $metadata );
```
